code/client/clrcore/External/VehicleWindowCollection.cs

```diff
@@ -48,14 +49,48 @@ internal VehicleWindowCollection(Vehicle owner)
 			}
 		}
 
+		public VehicleWindow[] GetAllWindows()
+		{
+			return
+				Enum.GetValues(typeof(VehicleWindowIndex)).Cast<VehicleWindowIndex>().Where(HasWindow).Select(windowIndex => this[windowIndex]).ToArray();
+		}
+
 		public bool AreAllWindowsIntact
 		{
 			get { return API.AreAllVehicleWindowsIntact(_owner.Handle); }
 		}
 
 		public void RollDownAllWindows()
 		{
-			API.RollDownWindows(_owner.Handle);
+			foreach(VehicleWindow vehicleWindow in this.GetAllWindows())
+			{
+				vehicleWindow.RollDown();
+			}
+		}
+
+		public void RollUpAllWindows()
+		{
+			foreach (VehicleWindow vehicleWindow in this.GetAllWindows())
+			{
+				vehicleWindow.RollUp();
+			}
+		}
+
+		public bool HasWindow(VehicleWindowIndex window)
```

Comment:
You probably want to cover more bones, this is a list of some bone names which might be interested https://hastebin.com/feyaqifaqe.txt you can find some of these in vehicles with a lot of windows like bus and so on..

Reply:
I guess [these ](https://github.com/citizenfx/fivem/blob/30e422b392d835f2c4a499b1f316b59ba4278bc4/code/client/clrcore/External/VehicleWindowCollection.cs#L15) extra windows are among these

Reply:
The vehicle `bus` has more windows like `window_lm1` to `window_lm6` and `window_rm1` to `window_rm5`, but only `window_lf`, `window_lr` and `window_rr` are "working" windows.
If you find a vehicle where the extra windows works, then i can try to find the bones.